addons/pos_restaurant/static/src/overrides/models/pos_store.js

```diff
@@ -105,6 +105,9 @@ patch(PosStore.prototype, {
                 } finally {
                     this.tableSyncing = false;
                 }
+            } else if (this.get_order()?.finalized && this.previousScreen !== "PaymentScreen") {
```

Comment:
In a way, I'm not even sure we need to check the previous screen if the order is paid. We should just go back to the floor_screen and unset the order.

Reply:
@davidmonnom tried this, but the problem is that you get kicked out of your receipt screen on the legitimate device in that case